gensim/models/tfword2vec.py

```diff
@@ -0,0 +1,206 @@
+
+    def intersect_word2vec_format(self, *args, **kwargs):
+        raise Exception("Cannot call a gensim training method on a tf trained model")
+
+
+class Options(object):
+    """Options class that doesn't use FLAGS"""
+
+    def __init__(self, train_data=None, save_path=None, eval_data=None,
+                 embedding_size=200, epochs_to_train=15, learning_rate=0.025,
+                 num_neg_samples=25, batch_size=500, concurrent_steps=12,
+                 window_size=5, min_count=5, subsample=1e-3):
+        """
+        train_data: Training data. E.g., unzipped file http://mattmahoney.net/dc/text8.zip.
+
+        save_path: Directory to write the model.
+
+        eval_data: Analogy questions.
+
+        embedding_size: The embedding dimension size.
+
+        epochs_to_train: Number of epochs to train. Each epoch processes the training data once
+
+        learning_rate: Initial learning rate
+
+        batch_size: Numbers of training examples each step processes
+
+        concurrent_steps: The number of concurrent training steps.
+
+        window_size: The number of worlds to predict to the left and right of the target word.
+
+        min_count: The minimum number of word occurrences for it to be included in the vocabulary.
+
+        subsample: Subsample threshold for word occurrence. Words that appear with higher frequency
+                   will be randomly down-sampled. Set to 0 to disable.
+        """
+
+        if train_data is None:
+            raise ValueError("train_data must be specified.")
+
+        # Model options.
+
+        # Embedding dimension.
+        self.emb_dim = embedding_size
+
+        # Training options.
+
+        # The training text file.
+        self.train_data = train_data
+
+        # Number of negative samples per example.
+        self.num_samples = num_neg_samples
+
+        # The initial learning rate.
+        self.learning_rate = learning_rate
+
+        # Number of epochs to train. After these many epochs, the learning
+        # rate decays linearly to zero and the training stops.
+        self.epochs_to_train = epochs_to_train
+
+        # Concurrent training steps.
+        self.concurrent_steps = concurrent_steps
+
+        # Number of examples for one training step.
+        self.batch_size = batch_size
+
+        # The number of words to predict to the left and right of the target word.
+        self.window_size = window_size
+
+        # The minimum number of word occurrences for it to be included in the
+        # vocabulary.
+        self.min_count = min_count
+
+        # Subsampling threshold for word occurrence.
+        self.subsample = subsample
+
+        # Where to write out summaries.
+        self.save_path = save_path
+
+        # Eval options.
+
+        # The text file for eval.
+        self.eval_data = eval_data
+
+
+def modified_tfw2v_init(self, options, session):
+    self._options = options
```

Comment:
Is there a conflict with another variable? Or why are these attributes internal (~underscored, `_`)?